packages/boot/test/bootstrapTests/ec-contract-upgrade.test.ts

```diff
@@ -0,0 +1,601 @@
+};
+
+const latestQuestionKey = `published.committees.Economic_Committee.latestQuestion`;
+const lastOutcomeKey = `published.committees.Economic_Committee.latestOutcome`;
+const highPrioritySenderKey = 'highPrioritySenders';
+
+const PROPOSAL_INV = 'old_proposal_invitation';
+const VOTER_INV = 'old_voter_invitation';
+const NEW_VOTER_INV = 'new_voter_invitation';
+
+const getQuestionId = id => `propose-question-${id}`;
+const getVoteId = id => `vote-${id}`;
+
+export const makeZoeTestContext = async t => {
+  console.time('ZoeTestContext');
+  const swingsetTestKit = await makeSwingsetTestKit(t.log, undefined, {
+    configSpecifier: '@agoric/vm-config/decentral-main-vaults-config.json',
+  });
+
+  const { runUtils, storage } = swingsetTestKit;
+  console.timeLog('DefaultTestContext', 'swingsetTestKit');
+  const { EV } = runUtils;
+
+  await eventLoopIteration();
+
+  // We don't need vaults, but this gets the brand, which is checked somewhere
+  // Wait for ATOM to make it into agoricNames
+  await EV.vat('bootstrap').consumeItem('vaultFactoryKit');
+  console.timeLog('DefaultTestContext', 'vaultFactoryKit');
+
+  // has to be late enough for agoricNames data to have been published
+  const agoricNamesRemotes = makeAgoricNamesRemotesFromFakeStorage(storage);
+  console.timeLog('DefaultTestContext', 'agoricNamesRemotes');
+
+  console.timeEnd('DefaultTestContext');
+  const walletFactoryDriver = await makeWalletFactoryDriver(
+    runUtils,
+    storage,
+    agoricNamesRemotes,
+  );
+
+  const smartWallets = await Promise.all(
+    Object.values(wallets).map(async addr =>
+      walletFactoryDriver.provideSmartWallet(addr),
+    ),
+  );
+  const { fromCapData } = makeMarshal(undefined, slotToBoardRemote);
+
+  return {
+    ...swingsetTestKit,
+    storage,
+    smartWallets,
+    fromCapData,
+  };
+};
+const test = anyTest as TestFn<Awaited<ReturnType<typeof makeZoeTestContext>>>;
+
+test.before(async t => {
+  t.context = await makeZoeTestContext(t);
+});
+
+test.serial('normal running of committee', async t => {
+  const { advanceTimeTo, storage, smartWallets, fromCapData } = t.context;
+
+  const agoricNamesRemotes = makeAgoricNamesRemotesFromFakeStorage(storage);
+  const { econCommitteeCharter, economicCommittee, VaultFactory } =
+    agoricNamesRemotes.instance;
+  const { ATOM: collateralBrand, IST: debtBrand } = agoricNamesRemotes.brand;
+
+  // Accepting all invitations for original committee
+  await Promise.all(
+    smartWallets.map(w =>
+      w.executeOffer({
+        id: PROPOSAL_INV,
+        invitationSpec: {
+          source: 'purse',
+          instance: econCommitteeCharter,
+          description: 'charter member invitation',
+        },
+        proposal: {},
+      }),
+    ),
+  );
+
+  await Promise.all(
+    smartWallets.map(w => {
+      const invitationPurse = w.getCurrentWalletRecord().purses.find(p => {
+        return p.brand.toString().includes('Invitation');
```

Comment:
Don't depend on the string form of a brand for correctness. It's a label for debugging only.

```suggestion
        return p.brand === agoricNames.brand.Invitation;
```

Reply:
seems we cant make that comparison
![image](https://github.com/user-attachments/assets/1a7f4860-80aa-4433-9690-fdc04f6dedd5)

I've tried to find some work around to make the comparison but there doesn't seem to be any

Reply:
bleah... yeah... ok... I just made
 - #10167

so...

```js
        // @ts-expect-error XXX BoardRemote #10167
        return p.brand === agoricNames.brand.Invitation;
```

Reply:
its not just a ts/lint issue

the object are inherently different. here the objects logged:
![image](https://github.com/user-attachments/assets/27a16bdf-a2bd-4d73-8837-07bb47e79eb3)
these two objects are not equal

Reply:
What code produced that result?

`agoricNames.brand.Invitation` was sort of a sketch. Perhaps `agoricNamesRemotes.brand.Invitation` is the precise way to get at the object.

_(also, in the interest of accessibility etc., please avoid [images of text](https://www.w3.org/WAI/tutorials/images/textual/))_

Reply:
this is the gist of the code we are running

```js
  const agoricNamesRemotes = makeAgoricNamesRemotesFromFakeStorage(storage);
  const { Invitation } = agoricNamesRemotes.brand;
  smartWallets.map(w => {
      const invitationPurse = w.getCurrentWalletRecord().purses.find(p => {
        return p.brand === Invitation;
      });
      if (!invitationPurse) return null;
  });
```

Reply:
bleah. `getCurrentWalletRecord` uses a different marshaller than agoricNamesRemontes.

OK, use `brand.toString().includes('Invitation');` but please do include a comment...

```
   // TODO: manage brands by object identity #10167
```